java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1953,6 +1955,54 @@ public void updateSystemInfo(JsonElement jsonResult, String minionId) {
         });
     }
 
+    public boolean updateMaster(MinionServer minion, String master) {
+        boolean changed = false;
+
+        Optional<Server> proxy = ServerFactory.lookupProxyServer(master);
+        if (proxy.isPresent()) {
+                // the system is connected to a proxy
+                // check if serverPath already exists
+                if (! ServerFactory.findServerPath(minion, proxy.get()).isPresent()) {
+                    // proxy path does not exist -> create it
+                    Set<ServerPath> proxyPaths = ServerFactory.createServerPaths(minion, proxy.get(), master);
+                    minion.getServerPaths().clear();
+                    minion.getServerPaths().addAll(proxyPaths);  //.orElse(Collections.emptySet()));
```

Comment:
The `orElse` part is part of the reason why this is WIP?

Reply:
I think the else in this case is not needed

Reply:
The comment should be deleted, sorry for confusion.